### DISCUSSION TITLE
Capture user request with request.state

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [encode/starlette#2197] [Q&A] [answered]
I'm using FastAPI and want to store the client request during a user session. I know this can be achieved with an authentication framework. Is it possible to use request.state (https://www.starlette.io/requests/#other-state) to capture the users request? If so, is there an example?

**Comment by alex-oleshkevich** [comment]
What do you mean by "user" and "client" requests?

**Reply by dineshbvadhia** [reply]
A user sends a request from a device (the client). Thanks

**Reply by alex-oleshkevich** [reply]
Sorry, I still don't understand. The Request class contains all the client info.
Please give more details. Would be glad to help you.

**Reply by dineshbvadhia** [reply, marked as answer]
Solved with lifespan-state :)
